### PR TITLE
Feature/jam_ctrl_gas_classes

### PR DESCRIPTION
**Pull request by carlyjb**
An initial implementation of the gas mixing class, implementing the math from the blending pdf.
Does not have a test harness or tests.

### CODE REVIEW

**Review by ralphBellofattoSTX — COMMENTED** [review]
looking at your fill class,

looks good, your section about bleedoff is actually not necessary for this application.

if we can't mix to the specified mix in one go, we can mix as close as possible and then start a breathing cycle.

We should be able to reach the desired gas mix within 6 breathing cycles.

As far as errors are concerned, we should raise some sort of error condition back to the UI if we can't reach the target pressure, within 3% within 6 breathing cycles.

NOTE: the whole area of raising error conditions and what to do has not been fleshed out yet.

All that said, i see nothing here that would prevent merging...

File: jamventProto/jamctrllib/JamCtrlGasMix.cpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
+/*
+ * FILE: JamCtrlGasMix.cpp
+ * 
+ * LICENSE: This file is licensed under the GNU General Public License v3.0
+ * Permissions of this strong copyleft license are conditioned on making available complete
```
Comment (jwildhome) [inline jamventProto/jamctrllib/JamCtrlGasMix.cpp:5]: Good that we are adding header and copyright.  Just wonder if this is the right license for us.  Just a comment. Nothing to change.

File: jamventProto/jamctrllib/JamCtrlGasMix.cpp [hunk 2]
```diff
@@ -0,0 +1,126 @@
+/*
```
Comment (jwildhome) [inline jamventProto/jamctrllib/JamCtrlGasMix.cpp:1]: Carly,

I think this looks good.  Just some comments more on format and conventions.

So some of these parameters are output or input/output parameters.
I think is good doxygen convention to comment them is the following.

 * @param  - default is this is an input parameter
 * @param[out] - output parameter
 * @param[in, out] - input/output parameter
Says he who has been kind of sparse on this comments.

I don't think we don't need to pass all these pointers. Just pass the value for input, and use a reference for output or input/output.

const float * const oxyPercDesired_fracDec,
->
 * @param reqAirTopUpPressure_bar ...
const float oxyPercDesired_fracDec,

and
float * const reqAirTopUpPressure_bar)
->
 * @param[out] reqAirTopUpPressure_bar ...
float& reqAirTopUpPressure_bar

**Review by ralphBellofattoSTX — APPROVED** [review]
this looks good to me

**Pull request merged**